ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAlgorithm.scala

```diff
@@ -16,36 +16,26 @@
 */
 package org.apache.spark.ml.h2o.algos
 
-import java.io._
-
 import hex.Model
 import hex.genmodel.utils.DistributionFamily
-import org.apache.hadoop.fs.Path
 import org.apache.spark.annotation.{DeveloperApi, Since}
 import org.apache.spark.h2o._
 import org.apache.spark.ml.h2o.param.{H2OAlgoParams, H2OModelParams}
 import org.apache.spark.ml.param.ParamMap
 import org.apache.spark.ml.util._
 import org.apache.spark.ml.{Estimator, Model => SparkModel}
 import org.apache.spark.sql.types.StructType
-import org.apache.spark.sql.{Dataset, SQLContext}
+import org.apache.spark.sql.{Dataset, SparkSession}
 import water.Key
 import water.support.H2OFrameSupport
 
 import scala.reflect.ClassTag
-
 /**
   * Base class for H2O algorithm wrapper as a Spark transformer.
   */
 abstract class H2OAlgorithm[P <: Model.Parameters : ClassTag, M <: SparkModel[M] : ClassTag]
-(parameters: Option[P])
-(implicit hc: H2OContext, sqlContext: SQLContext)
   extends Estimator[M] with MLWritable with H2OAlgoParams[P] {
 
-  if (parameters.isDefined) {
```

Comment:
This is useless code -> we set up the parameters using `updateH2OParams` in fit call

Reply:
Also getParams and setParams were removed -> they vere exposing internal Parameters H2O object